src/moments.jl

```diff
@@ -85,14 +85,16 @@ end
 function varm(A::RealArray, w::AbstractWeights, M::RealArray, dim::Int;
                    corrected::DepBool=nothing)
     corrected = depcheck(:varm, corrected)
-    varm!(similar(A, Float64, Base.reduced_indices(axes(A), dim)), A, w, M,
+    T = promote_type(promote_type(eltype(A), eltype(w)), eltype(M))
```

Comment:
`promote_type` has a vararg method, no need to nest calls

Reply:
If only that were in its docstring :angry: